### ISSUE TITLE
Pyright support of ParamSpec might has bug in some situations

### ISSUE THREAD

**Issue by MapleCCC** [microsoft/pyright#2178]
**To Reproduce**
```python
from collections.abc import Callable
from typing import Any

from typing_extensions import ParamSpec

P = ParamSpec("P")

def canonicalize_func_args(func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs):
    # Omitted code to canonicalize the arguments
    return 1

def hash_func_args(func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs):
    canonicalized = canonicalize_func_args(func, *args, **kwargs)
    return hash(canonicalized)
```

Pyright complains that `Argument of type "Any" cannot be assigned to parameter "args" of type "P@canonicalize_func_args" in function "canonicalize_func_args` `Type "Any" is incompatible with ParamSpec "P"`.

**Expected behavior**
Pyright should report no error.

**VS Code extension or command-line**
Pyright command-line 1.1.161

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.162, which I just published. It will also be in the next release of pylance.

**Comment by MapleCCC** [comment]
@erictraut Dear Eric, thank you for your kind help and productive fix. The problem is resolved perfectly by the new release.

However, there seems to be some new glitches in handling ParamSpec, introduced by the new release. I opened another issue (#2196) to track that.
